File: Baekjoon/Greedy/16435.py
# https://www.acmicpc.net/problem/16435

import sys
input = sys.stdin.readline

cnt, n = map(int, input().split())
fruits = list(map(int, input().split()))
fruits.sort()

for i in fruits:
    if i <= n:
        n += 1
        
print(n)

File: Baekjoon/Greedy/16435.cpp
// https://www.acmicpc.net/problem/16435

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n, l;
    vector<int> a;

    cin >> n >> l;
    a.resize(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    sort(a.begin(), a.end());
    for (int num : a) {
        if (l >= num) {
            l++;
        } else {
            break;
        }
    }
    cout << l << '\n';

    return 0;
}